lisa/sut_orchestrator/libvirt/ch_platform.py

```diff
@@ -0,0 +1,173 @@
+        assert isinstance(node.capability.core_count, int)
+        vcpu_count = node.capability.core_count
+        vcpu.text = str(vcpu_count)
+
+        os = ET.SubElement(domain, "os")
+
+        os_type = ET.SubElement(os, "type")
+        os_type.text = "hvm"
+
+        os_kernel = ET.SubElement(os, "kernel")
+        os_kernel.text = node_context.firmware_path
+
+        devices = ET.SubElement(domain, "devices")
+
+        console = ET.SubElement(devices, "console")
+        console.attrib["type"] = "pty"
+
+        console_target = ET.SubElement(console, "target")
+        console_target.attrib["type"] = "serial"
+        console_target.attrib["port"] = "0"
+
+        network_interface = ET.SubElement(devices, "interface")
+        network_interface.attrib["type"] = "network"
+
+        network_interface_source = ET.SubElement(network_interface, "source")
+        network_interface_source.attrib["network"] = "default"
+
+        network_model = ET.SubElement(network_interface, "model")
+        network_model.attrib["type"] = "virtio"
+
+        network_driver = ET.SubElement(network_interface, "driver")
+        network_driver.attrib["queues"] = str(vcpu_count)
+        network_driver.attrib["iommu"] = "on"
+
+        self._add_virtio_disk_xml(
+            node_context,
+            devices,
+            node_context.os_disk_file_path,
+            vcpu_count,
+        )
+
+        self._add_virtio_disk_xml(
+            node_context,
+            devices,
+            node_context.cloud_init_file_path,
+            vcpu_count,
+        )
+
+        xml = ET.tostring(domain, "unicode")
+        return xml
+
+    def _stop_and_delete_vm(
+        self,
+        environment: Environment,
+        log: Logger,
+        qemu_conn: libvirt.virConnect,
+        node: Node,
+    ) -> None:
+        super()._stop_and_delete_vm_flags(
+            environment,
+            log,
+            qemu_conn,
+            node,
+            0,  # ch driver currently doesn't support any flags
+        )
+
+    def _create_domain_and_attach_logger(
+        self,
+        libvirt_conn: libvirt.virConnect,
+        domain: libvirt.virDomain,
+        node_context: NodeContext,
+    ) -> None:
+        domain.createWithFlags(0)
+
+        assert node_context.console_logger
+        node_context.console_logger.attach(
+            libvirt_conn, domain, node_context.console_log_file_path
+        )
+
+    # Create the OS disk.
+    def _create_node_os_disk(
+        self, environment: Environment, log: Logger, node: Node
+    ) -> None:
+        node_context = get_node_context(node)
+        if node_context.os_disk_base_file_fmt == DiskImageFormat.QCOW2:
+            self.host_node.tools[QemuImg].convert(
+                "qcow2",
+                node_context.os_disk_base_file_path,
+                "raw",
+                node_context.os_disk_file_path,
+            )
+        else:
+            self.host_node.shell.spawn(
```

Comment:
This function doesn't wait for the process to complete. It returns a process object.

Reply:
Why call spawn here? The node.execute is the right layer to call.

Reply:
Changed to execute().